### PR TITLE
Implement attempts functionality for PDO-backend engines

### PR DESCRIPTION
**Pull request by robertscherer**
Refs #43 

We have implemented the attempts functionality in a way, that when adding a Job to the Queue, we can define the maximum number of attempts that can fail for a job before it is being released / discarded.

With `attempts_delay` we can specify how long we wait after a failed attempt before retrying.

### CODE REVIEW

File: docs/defining-jobs.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Jobs are simply PHP callables. Job callables should be available to the worker p
 
 Job callables receive a `Job` instance (not to be confused with your own own jobs), which is a wrapper around the metadata for a given job. The two useful methods of this `Job` instance are:
 
-- `attempts()`: Contains the number of times a job has been attempted. This value may be incorrect, depending upon the semantics of your chosen backend.
+- `attempts()`: Contains the number of attempts a job has has left before being discarded.
```
Comment (josegonzalez) [inline docs/defining-jobs.md]: Do all backends implement this?

**Comment by josegonzalez** [comment]
Have you tested this in prod?

**Comment by robertscherer** [comment]
@josegonzalez Not yet, but we're preparing for that. My understanding is, since attempts weren't implemented yet in the PdoEngine, this shouldn't affect existing behavior.

**Comment by josegonzalez** [comment]
Released in 0.0.3.

**Pull request merged**